streams/src/main/java/org/apache/kafka/streams/kstream/internals/StreamsTopologyGraphImpl.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.processor.internals.InternalTopologyBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
+class StreamsTopologyGraphImpl implements StreamsTopologyGraph {
+
+    private static final Logger LOG = LoggerFactory.getLogger(StreamsTopologyGraphImpl.class);
+    private StreamGraphNode previousNode;
+    private final AtomicInteger nodeIdCounter = new AtomicInteger(0);
+    public static final String TOPOLOGY_ROOT = "root";
```

Comment:
`TOPOLOGY_ROOT` seems not used outside of the class?

Reply:
needed for follow-on PR